gempy/assets/geographic.py

```diff
@@ -0,0 +1,562 @@
+
+    def get_orientation(self):
+        """Get orientation (dip_direction, dip) for points in all point set"""
+        if "normal" not in dir(self):
+            self.plane_fit()
+
+        # calculate dip
+        self.dip = np.arccos(self.normal[2]) / np.pi * 180.
+
+        # calculate dip direction
+        # +/+
+        if self.normal[0] >= 0 and self.normal[1] > 0:
+            self.dip_direction = np.arctan(self.normal[0] / self.normal[1]) / np.pi * 180.
+        # border cases where arctan not defined:
+        elif self.normal[0] > 0 and self.normal[1] == 0:
+            self.dip_direction = 90
+        elif self.normal[0] < 0 and self.normal[1] == 0:
+            self.dip_direction = 270
+        # +-/-
+        elif self.normal[1] < 0:
+            self.dip_direction = 180 + np.arctan(self.normal[0] / self.normal[1]) / np.pi * 180.
+        # -/-
+        elif self.normal[0] < 0 <= self.normal[1]:
+            self.dip_direction = 360 + np.arctan(self.normal[0] / self.normal[1]) / np.pi * 180.
+
+    #    elif normal_vec[1] == 0:
+    #        return 90
+
+    def stereonet(self):
+        """Create stereonet plot of plane pole and half circle for this point set"""
+        import matplotlib.pyplot as plt
+        import mplstereonet
+
+        fig = plt.figure()
+        ax = fig.add_subplot(111, projection='stereonet')
+
+        # ax.plane(dip_dirs, dips, 'g-', linewidth=0.5)
+        ax.pole(self.dip_direction - 90, self.dip, 'gs', markersize=4)
+        ax.plane(self.dip_direction - 90, self.dip, 'g', markersize=4)
+
+        # ax.rake(strike, dip, -25)
+        ax.grid()
+
+    def minmax(self):
+        """Get minimum and maximum values of points in point set (e.g. to determine surrounding box)"""
+        point_array = np.empty((len(self.points), 2))
+
+        for i, p in enumerate(self.points):
+            point_array[i] = (p.x, p.y)
+
+        self.min = np.min(point_array, axis=0)
+        self.max = np.max(point_array, axis=0)
+
+
+class GeoTiffgetValue(object):
+    """Determin a value from a GeoTiff
+
+    Required, for example, to determine z-values in GeoTiffs of DEMs.
+
+    Note: at this stage, the value is taken from Raster Band 1.
+
+    Credits to entry on stackoverflow:
+    http://stackoverflow.com/questions/13439357/extract-point-from-raster-in-gdal
+
+    Args:
+        tifname (filename): path and filename to GeoTiff file
+    """
+
+    def __init__(self, tifname='test.tif'):
+        # open the GeoTiff raster file and its spatial reference
+        self.ds = gdal.Open(tifname)
+        sr_raster = osr.SpatialReference(self.ds.GetProjection())
+
+        # get the WGS84 spatial reference
+        sr_point = osr.SpatialReference()
+        sr_point.ImportFromEPSG(4326)  # WGS84
+
+        # coordinate transformation
+        self.ct = osr.CoordinateTransformation(sr_point, sr_raster)
+
+        # geotranformation and its inverse
+        gt = self.ds.GetGeoTransform()
```

Comment:
using rasterio can basically remove this whole class or reduce it to less than 10 lines of code